### PR TITLE
Collapsing of logic trees (3)

### PR DESCRIPTION
**Pull request by micheles**
A first test of the collapsing feature. More will follow.

### CODE REVIEW

File: openquake/hazardlib/tests/lt_test.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+</pointSource>''')
+
+
+def scaling_rates(srcs):
+    return [getattr(src, 'scaling_rate', 1.) for src in srcs]
+
+
+class CollapseTestCase(unittest.TestCase):
+    def setUp(self):
+        # setup a simple logic tree with 3 realizations
```
Comment (mmpagani) [inline openquake/hazardlib/tests/lt_test.py]: This schematic does not reflect the LT structure used.

File: openquake/hazardlib/tests/lt_test.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        # self.plot(mean, coll1)
+        assert scaling_rates(srcs) == [1.0, 0.5, 0.5, 1.0]
+        numpy.testing.assert_allclose(mean, coll1, atol=.1)
+
+        # compute the fully collapsed curve
+        self.bs0.collapsed = True
+        self.bs1.collapsed = True
+        coll2, srcs, weights = self.full_enum()
+        assert weights == [1]  # one rlz
+        # self.plot(mean, coll2)
```
Comment (mmpagani) [inline openquake/hazardlib/tests/lt_test.py]: leftover

File: openquake/hazardlib/tests/lt_test.py [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+        assert scaling_rates(srcs) == [0.4, 0.6, 0.5, 0.5]
+        numpy.testing.assert_allclose(mean, coll2, atol=.16)
```
Comment (mmpagani) [inline openquake/hazardlib/tests/lt_test.py]: I am assuming that here you compare the whole mean with the mean you get by collapsing just 1 uncertainty.

Comment (micheles) [reply openquake/hazardlib/tests/lt_test.py]: Yep!

File: openquake/hazardlib/tests/lt_test.py (hunk 2 above)
Comment (micheles) [reply openquake/hazardlib/tests/lt_test.py]: No, the comment is there on purpose, it is useful uncomment that line and look at the curves

**Pull request merged**